server/task/task-api/pom.xml

```diff
@@ -47,6 +47,10 @@
             <groupId>com.google.guava</groupId>
             <artifactId>guava</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.projectreactor</groupId>
+            <artifactId>reactor-core</artifactId>
```

Comment:
why did you need that? It looks like a violation of the "no reactor in public APIs" rule

Reply:
(I will drop that commit in a separate PR to not delay the fix)

Well because I follow the same pattern for reactive adoption than what we did for mailbox reactive mailbox listener.

What I believed was the 'state of the art' for progressive reactive code adoption.

This is needed:
 - So that a ReactiveTask can provide a default event method (blocker)
 - So that we can provide a TaskReactiveWrapper (not a blocker: can be moved to the executor)

>  It looks like a violation of the "no reactor in public APIs" rule

Given the above content maybe the rule should be:

 - No reactor Mono / Flux in public API interface signature
 - But it's use is ok in default methods, so we have the right to have it as part of dependencies of API maven modules

WDYT?

Reply:
> > It looks like a violation of the "no reactor in public APIs" rule
> 
> Given the above content maybe the rule should be:
> 
>     * No reactor Mono / Flux in public API interface signature
> 
>     * But it's use is ok in default methods, so we have the right to have it as part of dependencies of API maven modules
> 
> 
> WDYT?

Doing that always ends the same way: we put reactor types everywhere by mistake.

I propose to write a wrapping method and duplicates in every maven module that needs it: there's almost not logic in it so the risk of bugs it very thin, it's tested as part of higher level testsuite and it will disappear as we do the migration.

The cost/risk of the duplication is in my opinion less than the cost/risk of introducing a depency on a library we don't want to depend on

Reply:
> Doing that always ends the same way: we put reactor types everywhere by mistake.

You being so pessimistic about a rule as simple as `don't use reactor in API interfaces` that I believe everyone of us could easily enforce now that it is clearly established depresses me.

I don't see the added value of the maven dependency enforcement: it is as easy in my opinion to enforce `No reactor Mono / Flux in public API interface signature`

Now what's the associated risk? Turning Mono/Flux into Publisher in case of abuse is no hard work and a pretty easy refactoring. I have no doubt that we could easily cary it out.

Now when I compare the benefits (not writing run() for every task) to the exposure (low) and the consequences (limited) I don't see why I should refrain.

> I propose to write a wrapping method and duplicates in every maven module that needs it: there's almost not logic in it so the risk of bugs it very thin, it's tested as part of higher level testsuite and it will disappear as we do the migration.

Well if we are down to that I still prefer wrinting the .block call in every class.

> The cost/risk of the duplication is in my opinion less than the cost/risk of introducing a depency on a library we don't want to depend on

I see that we have a diverging analyses here.

Reply:
> I see that we have a diverging analyses here.

Because I did several cleanups in the past because of this accidental dependency and it was not always simple tasks.

It's not always as easy as changing a return type and it costed us much time in the past,

Even cases where it was simple, it was tedious.

Reply:
> Because I did several cleanups in the past because of this accidental dependency and it was not always simple tasks.

How established was the practice? How much do people care.

Have mentalities and awareness evolved since then?

Reply:
> Well if we are down to that I still prefer wrinting the .block call in every class.

I will do that tomorrow even if that outcome don't please me at all.

Reply:
> > Because I did several cleanups in the past because of this accidental dependency and it was not always simple tasks.
> 
> How established was the practice? How much do people care.
> 
> Have mentalities and awareness evolved since then?

I can't really tell, sorry